### PR TITLE
Update profile url workaround

### PR DESCRIPTION
**Pull request by ttannis**
### Summary of Changes

Passing `app.config['GET_PROFILE_URL']` into the `user_data` yields a json not serializable error in some environments. The source of this bug in in amundsencommon: https://github.com/lyft/amundsencommon/pull/36#discussion_r427602371.

To temporarily fix this issue, we are reintroducing some of the previous logic that was removed in 
https://github.com/lyft/amundsenfrontendlibrary/pull/440

We will file a follow up issue on the ideal fix for this case, which is to include `profile_url` when user metadata in ingested, and we can completely remove the `GET_PROFILE_URL` configuration from Amundsen, in favor of generating it on the backend.

### Tests

N/A, all code added in this PR is intended to be deleted once we complete the follow up ticket.

### Documentation

N/A no functionality added, but we will be documenting the follow up ticket.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes, including screenshots of any UI changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

**Comment by ttannis** [comment]
cc @verdan. I have to add some code back to this repo to avoid the issue I mentions in amundsencommon, and we will make plans to add `profile_url` to the metadata so that we can remove these workarounds.

File: amundsen_application/models/user.py [hunk 1]
```diff
@@ -1,15 +1,30 @@
-from typing import Dict
+from typing import Dict, Optional
 
 from amundsen_common.models.user import UserSchema, User
 from flask import current_app as app
 from marshmallow import ValidationError
 
 
+def _str_no_value(s: Optional[str]) -> bool:
```
Comment (feng-tao) [inline amundsen_application/models/user.py:8]: maybe a github issue todo to remove this workaround once metadata user object includes profile url as well?
cc @verdan

Comment (ttannis) [reply amundsen_application/models/user.py:8]: Will make sure we have an issue

**Comment by ttannis** [comment]
Followup: https://github.com/lyft/amundsen/issues/456

**Pull request merged**